screeps-game-api/src/constants.rs

```diff
@@ -979,3 +995,60 @@ impl TryFrom<Value> for ResourceType {
         })
     }
 }
+
+// TODO: reactions
+// TODO: boosts
+
+pub const PORTAL_UNSTABLE: u32 = 10 * 24 * 3600 * 1000;
+pub const PORTAL_MIN_TIMEOUT: u32 = 12 * 24 * 3600 * 1000;
+pub const PORTAL_MAX_TIMEOUT: u32 = 22 * 24 * 3600 * 1000;
+
+pub const POWER_BANK_RESPAWN_TIME: u32 = 50000;
+
+pub const INVADERS_ENERGY_GOAL: u32 = 100000;
+
+pub const SYSTEM_USERNAME: &str = "Screeps";
+
+pub const SIGN_PLANNED_AREA: &str =
+    "A new Novice or Respawn Area is being planned somewhere \
+     in this sector. Please make sure all important rooms are reserved.";
+
+pub const POWER_LEVEL_MULTIPLY: u32 = 1000;
+pub const POWER_LEVEL_POW: u32 = 2;
+pub const POWER_CREEP_SPAWN_COOLDOWN: u32 = 8 * 3600 * 1000;
+pub const POWER_CREEP_DELETE_COOLDOWN: u32 = 24 * 3600 * 1000;
+pub const POWER_CREEP_MAX_LEVEL: u32 = 25;
+pub const POWER_CREEP_LIFE_TIME: u32 = 5000;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "constants-serde", derive(Serialize, Deserialize))]
+pub enum PowerClass {
+    #[cfg_attr(feature = "constants-serde", serde(rename = "operator"))]
+    Operator,
+}
+
+// Serialize_repr / Deserialize_repr serialize/deserialize as numbers
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize_repr, Deserialize_repr)]
+#[repr(u32)]
+pub enum PowerType {
+    GenerateOps = 1,
+    OperateSpawn = 2,
+    OperateTower = 3,
+    OperateStorage = 4,
+    OperateLab = 5,
+    OperateExtension = 6,
+    OperateObserve = 7,
+    OperateTerminal = 8,
+    DisruptSpawn = 9,
+    DisruptTower = 10,
+    Shield = 12,
+    RegenSource = 13,
+    RegenMineral = 14,
+    DisruptTerminal = 15,
+    OperatePower = 16,
+    Fortify = 17,
+    OperateController = 18,
+    OperateFactory = 19,
+}
+
+// TODO: POWER_INFO
```

Comment:
Wasn't there anything about tunnels also? I'm uncomfortable bumping the date at the top of the file without catching everything that was updated in the constants.
Also, how hard could it be to make a macro that uses the `js` file that describes the constants directly to generate this file?

Reply:
I think the only tunnel thing was `CONSTRUCTION_COST_ROAD_WALL_RATIO`, which I added above.

But I agree with the concerns here - I'll either implement the rest of them or make a summary TODO and TODO issue.

Reply:
With automated stuff, I bet we could do something for all the "plain" constants which are just numbers. But I think it might be hard translating to enums like ResourceType, PowerType, and such. I mean I haven't even figured out how we want to translate reactions nor power info - I _think_ functions which match on arguments and return the result would be the best way, but that might not be true.

Reply:
Maybe later...